src/commands/edge_app.rs

```diff
@@ -0,0 +1,250 @@
+                .is_empty()
+        {
+            edge_app_json.remove("id");
+        }
+
+        edge_app_json.remove("created_at");
+        edge_app_json.remove("created_by");
+        edge_app_json.remove("updated_at");
+        edge_app_json.remove("permissions");
+
+        let mut payload = HashMap::<String, String>::new();
+        // for now all values are fields in the manifest are required to be non-empty.
+        // if we change that we will need to have a list of required non-empty values.
+        debug!("Edge app headers: ");
+        for (k, v) in edge_app_json {
+            debug!("{k}: {v:?}");
+            let value = v.as_str().ok_or(CommandError::MissingField)?;
+            if value.is_empty() {
+                return Err(CommandError::InvalidManifestValue(k.to_string()));
+            };
+
+            payload.insert(k.to_owned(), value.to_owned());
+        }
+
+        let mut headers = HeaderMap::new();
+        headers.insert("Prefer", "return=representation".parse()?);
+
+        let response = self
+            .authentication
+            .build_client()?
+            .post(url)
+            .headers(headers)
+            .json(&payload)
+            .send()?;
+
+        if response.status() != StatusCode::CREATED {
+            return Err(CommandError::WrongResponseStatus(
+                response.status().as_u16(),
+            ));
+        }
+
+        // I think there is no need to check for size - this vector should always be size 1.
+        // We can let it crash in run time otherwise.
+        let updated_manifests: Vec<EdgeAppManifest> = serde_json::from_str(&response.text()?)?;
+
+        // overwrite manifest file now that we have id and extra fields.
+        let yaml = serde_yaml::to_string(&updated_manifests[0])?;
+        let input = File::create(path)?;
+        write!(&input, "{yaml}")?;
+
+        Ok(updated_manifests[0].clone())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::authentication::Config;
+
+    use envtestkit::lock::lock_test;
+    use envtestkit::set_env;
+    use httpmock::Method::POST;
+    use httpmock::MockServer;
+    use std::ffi::OsString;
+    use tempdir::TempDir;
+
+    #[test]
+    fn test_edge_app_init_should_create_screenly_yml() {
+        let tmp_dir = TempDir::new("test").unwrap();
+        let _lock = lock_test();
+        let _test = set_env(OsString::from("HOME"), tmp_dir.path().to_str().unwrap());
+        fs::write(tmp_dir.path().join(".screenly").to_str().unwrap(), "token").unwrap();
+
+        // init should make no requests so it's whatever for server url
+        let config = Config::new("asdf".to_owned());
+        let authentication = Authentication::new_with_config(config);
+        let command = EdgeAppCommand::new(authentication);
+
+        let p = tmp_dir.path().join("screenly.yml");
+        assert!(command.init(Path::new(p.to_str().unwrap())).is_ok());
+
+        assert_eq!(
+            EdgeAppManifest::default(),
```

Comment:
Let's be precise here meaning raw string

```rust
#r"name:
    - some_name
description:
    - ....
"#
```

or load yaml from this string.

Reply:
what's the purpose of that? but fine.

Reply:
Usually test with precise strings is more readable and is determined. 
Imagine you make a mistake in the result of default() function and response would include your mistake - in that way test will pass, but there will be an error.

But comparing it always to exact response would catch that error.